packages/frontend/src/stores/messagelist.ts

```diff
@@ -53,11 +53,12 @@ const defaultState = () =>
   }) as MessageListState
 
 /*
- * A hook to read a portion of messages(a view) for a given chat. It creates a store(MessageListStore)
+ * A hook to read a portion of messages(a view) for a given chat. It uses the passed store(MessageListStore)
  * for the given chat and account and loads messages on it. It always has a maximum specified number
- * of messages as per PAGE_SIZE constant.
+ * of messages as per PAGE_SIZE constant. The store is also used in useDraft to avoid redundant data loading.
  */
 export function useMessageList(
+  store: MessageListStore,
   accountId: number,
   chatId: number
```

Comment:
With this function signature it is now possible to provide a MessageListStore whose `chatId` and `accountId` are different from that of `store`. I think we should obtain them from the `store` and remove the parameters.